### PR TITLE
Remove Deprecated code from Core component

### LINKED ISSUE #4983: Remove 3.x series deprecated code
We are approaching the release of Solidus 4.0. 

That version will be as similar as possible to the last of the 3.x series to reduce the number of breaking changes we introduce. One of the major differences will be removing all the deprecated code. 

Assuming people stopped using the code deprecated, this will be the part of the update that won't break any application. 

Here's the list of PRs that remove deprecated code:

- https://github.com/solidusio/solidus/pull/5026
- https://github.com/solidusio/solidus/pull/4890
- https://github.com/solidusio/solidus/pull/5019
- https://github.com/solidusio/solidus/pull/5020
- https://github.com/solidusio/solidus/pull/5021
- https://github.com/solidusio/solidus/pull/5022
- https://github.com/solidusio/solidus/pull/5023
- https://github.com/solidusio/solidus/pull/5024
- https://github.com/solidusio/solidus/pull/4989

### PR DESCRIPTION
**Pull request by kennyadsl**
This PR is part of [3.x Deprecations Removal](https://github.com/solidusio/solidus/issues/4983).

## Summary

This PR removes all the code deprecated in the Core component. Please refer to each commit to better understand what has been removed and their replacements. 

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: core/lib/spree/bus.rb [hunk 1]
```diff
@@ -5,16 +5,9 @@ module Spree
...
   Bus = Omnes::Bus.new
   def Bus.publish(event, **kwargs)
-    if Spree::Config.use_legacy_events
-      Spree::Event.fire(event, **kwargs)
-    else
-      # Override caller_location to point to the actual event publisher
-      super(event, **kwargs, caller_location: caller_locations(1)[0])
-    end
+    # Override caller_location to point to the actual event publisher
+    super(event, **kwargs, caller_location: caller_locations(1)[0])
```
Comment (waiting-for-dev) [inline core/lib/spree/bus.rb:11]: We can remove the `def Bus.publish` override altogether. As we won't override it, there's no need to change the default caller location.

Comment (kennyadsl) [reply core/lib/spree/bus.rb]: Done!

**Review by waiting-for-dev — APPROVED** [review]
Great job!! :rocket:

File: core/spec/models/spree/fulfilment_changer_spec.rb [hunk 2]
```diff
@@ -81,9 +81,8 @@
       )
     end
 
-    it "defaults to true with a deprecation warning" do
-      expect(Spree::Deprecation).to receive(:warn).with(/Not passing `track_inventory` to `Spree::FulfilmentChanger` is deprecated./)
-      expect(shipment_splitter.track_inventory).to be true
+    it "raises an error" do
+      expect { subject }.to raise_error(ArgumentError)
```
[the remark this answers is not in the document]
Comment (kennyadsl) [reply core/spec/models/spree/fulfilment_changer_spec.rb]: Done!

File: core/app/models/spree/fulfilment_changer.rb [hunk 3]
```diff
@@ -17,27 +17,19 @@ module Spree
   # @attr [Integer] quantity How many units we want to move
   #
   class FulfilmentChanger
-    TRACK_INVENTORY_NOT_PROVIDED = Object.new.freeze
```
[the remark this answers is not in the document]
Comment (kennyadsl) [reply core/app/models/spree/fulfilment_changer.rb:20]: @waiting-for-dev https://github.com/solidusio/solidus/pull/5028, WDYT?

[the remark this answers is not in the document]
Comment (waiting-for-dev) [reply core/app/models/spree/fulfilment_changer.rb:20]: :+1:

**Review by adammathys — APPROVED** [review]

**Pull request merged**